src/actions/AuthActions.js

```diff
@@ -1,67 +1,105 @@
-    dispatch({
-        type: LOGIN_AUTH_ERROR,
-        payload: err
-    });
-};
+    let {type, token} = await Facebook.logInWithReadPermissionsAsync(appID, faceBookOptions);
 
-export const loginSuccess = (dispatch, token) => {
-    // TODO token arg will be used when we add FB Auth
-    let tokenToSend = token;
+    if (type === 'success') {
+        let {data: {email, name, picture: {data: {url}}}} = await axios.get(`https://graph.facebook.com/me?fields=id,name,email,picture&access_token=${token}`);
+        // we now have access to email, name, and their profile picture
 
-    if (!tokenToSend) {
-        tokenToSend = 'Manually Generated Token'
-    }
+        await AsyncStorage.setItem(tokens.USER_TOKEN, token);
 
-    dispatch({
-        type: LOGIN_SUCCESS,
-        payload: tokenToSend
-    });
+        dispatch({
+            type: FACEBOOK_LOGIN_SUCCESS,
+            payload: token
+        })
+    } else {
+        dispatch({
+            type: FACEBOOK_LOGIN_FAILURE
+        })
+    }
 };
 
-export const loginUser = ({email, password}) => {
-    return (dispatch) => {
-        firebase.auth().signInWithEmailAndPassword(_.trim(email), _.trim(password))
-            .then(user => {
-                loginSuccess(dispatch);
-            })
-            .catch(err => {
-                loginAuthError(dispatch, err);
-            })
+export const logUserIntoFacebook = () => {
+
+    return async (dispatch) => {
+        let token = await AsyncStorage.getItem(tokens.USER_TOKEN);
+
+        if (token) {
+            checkAndSetToken(token);
+        } else {
+            attemptFacebookLogin(dispatch);
+        }
     }
 };
 
-export const logoutUser = () => {
-    return (dispatch) => {
-        firebase.auth().signOut()
-            .then((user) => {
-                console.log("Logged the user out", user)
+export const checkAndSetToken = (passedToken) => {
+
+    return async (dispatch) => {
+        let token = passedToken;
 
-                //TODO add a trigger here to route back to login screen
+        if (!token) {
+            token = await AsyncStorage.getItem(tokens.USER_TOKEN);
+        }
+
+        if (token) {
+            if (token === tokens.GUEST) {
+                dispatch({
+                    type: GUEST_TOKEN_SET,
+                    payload: {token, appReady: true}
+                });
+            } else {
+                dispatch({
+                    type: FACEBOOK_LOGIN_SUCCESS,
+                    payload: {token, appReady: true}
+                })
+            }
+        } else {
+            dispatch({
+                type: APP_READY
             })
+        }
     }
-}
\ No newline at end of file
+};
+
+export const setGuestToken = () => {
+
+    return async (dispatch) => {
+        await AsyncStorage.setItem(tokens.USER_TOKEN, tokens.GUEST);
+
+        dispatch({
+            type: GUEST_TOKEN_SET,
+            payload: tokens.GUEST
+        });
+    };
+};
```

Comment:
This is called when we say "continue as guest" in the `LoginScreen`. Basically, since all the other screens depend on a token, we set a `guest` one so that it's not `null`.